src/modules/sensors/sensors.cpp

```diff
@@ -159,6 +161,22 @@ bool Sensors::init()
 
 int Sensors::parameters_update()
 {
+	if (_hil_enabled) {
+#if defined(CONFIG_SENSORS_VEHICLE_GPS_POSITION)
+		InitializeVehicleGPSPosition();
+#endif // CONFIG_SENSORS_VEHICLE_GPS_POSITION
+
+#if defined(CONFIG_SENSORS_VEHICLE_AIR_DATA)
+		InitializeVehicleAirData();
+#endif // CONFIG_SENSORS_VEHICLE_AIR_DATA
+
+#if defined(CONFIG_SENSORS_VEHICLE_MAGNETOMETER)
+		InitializeVehicleMagnetometer();
+#endif // CONFIG_SENSORS_VEHICLE_MAGNETOMETER
+
+		_fakeSensors.update(_failureDetector);
+	}
+
```

Comment:
shall we move this block to be after this block at L256 ?
```
#if defined(CONFIG_SENSORS_VEHICLE_AIR_DATA)
	InitializeVehicleAirData();
#endif // CONFIG_SENSORS_VEHICLE_AIR_DATA
#if defined(CONFIG_SENSORS_VEHICLE_GPS_POSITION)
	InitializeVehicleGPSPosition();
#endif // CONFIG_SENSORS_VEHICLE_GPS_POSITION
#if defined(CONFIG_SENSORS_VEHICLE_MAGNETOMETER)
	InitializeVehicleMagnetometer();
#endif // CONFIG_SENSORS_VEHICLE_MAGNETOMETER
#if defined(CONFIG_SENSORS_VEHICLE_OPTICAL_FLOW)
	InitializeVehicleOpticalFlow();
#endif // CONFIG_SENSORS_VEHICLE_OPTICAL_FLOW
```

Reply:
Unfortunately no, because this block should work in `amed_`  case and be able to change status of sensors during a fly.
```
	if (_armed) {
		return 0;
	}
```